.github/workflows/ci.yml

```diff
@@ -188,14 +188,49 @@ jobs:
         files: mockito-integration-tests/android-tests/build/reports/coverage/android-tests/debug/connected/report.xml
         fail_ci_if_error: true
 
+  #
+  # GraalVM native tests job
+  #
+  graalvm:
+    runs-on: ubuntu-latest
+    if: "! contains(toJSON(github.event.commits.*.message), '[skip ci]')"
+
+    steps:
+    - name: 1. Check out code
+      uses: actions/checkout@v5
+      with:
+        fetch-depth: '0'
+
+    - name: 2. Setup GraalVM
+      uses: graalvm/setup-graalvm@7f488cf82a3629ee755e4e97342c01d6bed318fa
+      with:
+        java-version: "21"
+        distribution: "graalvm"
+        cache: gradle
+
+    - name: 3. Run GraalVM native tests with metadata
+      run: ./gradlew :mockito-integration-tests:graalvm-tests:nativeTest -PenableGraalTracingAgent --stacktrace --scan
+
+    - name: 4. Verify GraalVM metadata generation
+      run: |
+        REFLECT_CONFIG="mockito-integration-tests/graalvm-tests/src/test/resources/META-INF/native-image/org.mockito/graalvm-tests/reflect-config.json"
```

Comment:
Can we not write this as a JUnit test? I prefer JUnit assertions over a Bash verification. If that ain't possible, then c'est ca.

Reply:
Ah okay I see. All right let's keep it like this.

Reply:
That's true – I don't think there is a straightforward way to move this test into junit.